src/server/bots/bridgeless/txidSubmissionService.ts

```diff
@@ -0,0 +1,203 @@
+  ) {
+    return clean.blockHeight
+  } else {
+    return 0
+  }
+}
+
+const ZANO_RPC_URL = 'http://37.27.100.59:10500'
+
+const asZanoHeight = asObject({
+  height: asNumber,
+  status: asValue('OK')
+})
+const getZanoChainHeight = async (): Promise<number> => {
+  const json = await doFetch(`${ZANO_RPC_URL}/getheight`)
+  const clean = asZanoHeight(json)
+  return clean.height
+}
+
+const asZanoTransactionHeight = asObject({
+  // id: 0,
+  // jsonrpc: '2.0',
+  result: asObject({
+    status: asValue('OK'),
+    tx_info: asObject({
+      // amount: 18999000000000,
+      // attachments: [
+      //   {
+      //     details_view: '',
+      //     short_view:
+      //       '0feef5e2ea0e88b592c0a0e6639ce73e12ea9b3136d89464748fcb60bb6f18f5',
+      //     type: 'pub_key'
+      //   }
+      // ],
+      // blob: 'ARMBgKCUpY0dBBoAAAAAAAAAABoCAAAAAAAAABoKAAAAAAAAABoPAAAAAAAAACVA4FRLH',
+      // blob_size: 6794,
+      // extra: [
+      //   {
+      //     details_view: '',
+      //     short_view:
+      //       '0feef5e2ea0e88b592c0a0e6639ce73e12ea9b3136d89464748fcb60bb6f18f5',
+      //     type: 'pub_key'
+      //   }
+      // ],
+      // fee: 1000000000,
+      // id: 'a6e8da986858e6825fce7a192097e6afae4e889cabe853a9c29b964985b23da8',
+      // ins: [
+      //   {
+      //     amount: 1000000000000,
+      //     global_indexes: [0, 2, 12, 27],
+      //     htlc_origin: '',
+      //     kimage_or_ms_id:
+      //       '2540e0544b1fed3b104976f803dbd83681335c427f9d601d9d5aecf86ef276d2',
+      //     multisig_count: 0
+      //   }
+      // ],
+      keeper_block: asNumber
+      // object_in_json: 'ewogICJ2ZXJzaW9uIjogMSwgCiAgInZpbiI6IFsgewogICAgIC',
+      // outs: [
+      //   {
+      //     amount: 9000000000,
+      //     global_index: 0,
+      //     is_spent: false,
+      //     minimum_sigs: 0,
+      //     pub_keys: [
+      //       '7d0c755e7e24a241847176c9a3cf4c970bcd6377018068abe6fe4535b23f5323'
+      //     ]
+      //   }
+      // ],
+      // pub_key:
+      //   '0feef5e2ea0e88b592c0a0e6639ce73e12ea9b3136d89464748fcb60bb6f18f5',
+      // timestamp: 1557345925
+    })
+  })
+})
+const getZanoTransactionHeight = async (txid: string): Promise<number> => {
+  const body = {
+    id: 0,
+    jsonrpc: '2.0',
+    method: 'get_tx_details',
+    params: {
+      tx_hash: txid
+    }
+  }
+  const opts = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  }
+  const json = await doFetch(`${ZANO_RPC_URL}/json_rpc`, opts)
+  const clean = asZanoTransactionHeight(json)
+  return Math.max(clean.result.tx_info.keeper_block, 0)
+}
+
+export const chainUtils: Record<
+  string,
+  {
+    getRequiredConfirmations: () => Promise<number>
+    getChainHeight: () => Promise<number>
+    getTxHeight: (txid: string) => Promise<number>
+  }
+> = {
+  // bitcoin
+  '0': {
+    getRequiredConfirmations: async () => await getRequiredConfirmations('0'),
```

Comment:
`getRequiredConfirmations` doesn't have to be part of this object. make it a static func that just gets called with the chainId